### DISCUSSION TITLE
How to use "New boundary conditions for the oscillating velocity from far fields"

### DISCUSSION THREAD

**Discussion by 101709** [mdolab/dafoam#711] [General]
I also want to ask the question of unsteady optimization, I saw that DAfoam implemented "New boundary conditions for the oscillating velocity from far fields" and gave an example: "myPatch
{
type             varyingVelocityInletOutlet;
U0               10;
URate-0.2;
flowComponent     0;
normalComponent   1;
alpha0           0;
alphaRate 0.1;
}"
But I don't know how to modify the file is right, I modified "0 _orig under" file "U", the result is an error "Unknown patchField type varyingVelocityInletOutlet for patch type patch". What should I do with this, please? I don't see any tutorials on the website? Thank you for answering the question！
[pitch.zip](https://github.com/user-attachments/files/17894113/pitch.zip)

**Comment by 101709** [comment]
In this case, for example, the Angle of attack should not be adjustable as a design variable, but according to pyDAFoam, "if no alpha is added in DVGeo.addGeoDVGlobal, we can NOT use parallelToFlow.". But lift should always be perpendicular to the incoming flow, and drag should always be parallel to the incoming flow. So how do I set up optimization in the case ?Could you please tell me which areas need to be modified? Or is there a complete and operational case for me to learn from? I really need your help, I have tried many times but haven't been able to solve the problem well. Thank you so much!

**Comment by friedenhe** [comment]
Please elaborate on what you try to do with the varyingVelocityInletOutlet BC. I feel that this BC is probably not what you want to use.

**Reply by 101709** [reply]
Ok, let me be a little more specific:
Objective: To simulate that the airfoil/wing has better lift and drag characteristics under periodic changes in far-field conditions.
See the case: CurvedCubeHexMesh folder under the 0.unsteady U file
![image](https://github.com/user-attachments/assets/c99508ba-e93b-48d5-bc85-58555b1ba0b5)
Question 1: These boundary condition types are not available in openfoam, can I set them in U file in dafoam?
Question 2:
The direction of lift and drag in “runTests_DAPisoFoam“ is fixed because the Angle of attack of the incoming stream does not change. If I change ”component“ to 1, that is, y direction, in which case the velocity in x direction remains unchanged and the velocity in y direction oscillates, the Angle of attack will change. Then, can the directions of lift and drag be set to ”normalToFlow“ or ”parallelToFlow“? Because the Angle of attack is not a design variable.  If possible, how should ”addGeoDVGlobal("alpha", [alpha0], alpha, lower=-10.0, upper=10.0, scale=1.0)“ be modified?
Question 3: Can the force coefficient be modified at this case? If possible, how to modify it, because the speed value of the incoming stream is changing.
![image](https://github.com/user-attachments/assets/803c072e-e62c-4549-b6ae-6330c204b33c)
Question 4: what is the difference between varyingVelocity and varyingVelocityInletOutlet?
I am looking forward to your detailed reply and hope to have some cases. Thank you very much for your answer, which is very important to me. I wish you a smooth scientific research.

**Comment by friedenhe** [comment]
Overall, I believe varyingVelocityInletOutlet, etc should be the BC you should use.
1. Yes, it is DAFoam-specific implementations, and the source codes are: https://github.com/mdolab/dafoam/tree/main/src/adjoint/boundaryConditions
2. No, you have to use fixedDirection for unsteady objFuncs. normaltoFlow and parallelToFlow do not work for unsteady CD and CL.
3. No... Changing the scaling for CD and CL during unsteady simulations is not implemented. I suggest you use force in optimization, instead of force coefficient.
4. Check the difference between fixedValue and inletOutlet BCs in OpenFOAM.

**Reply by 101709** [reply]
Your answer is very clear. Let me confirm again, do I only need to modify the `inlet type` in `0/U`? And how do you choose the type of `outlet`? Because I've tried it before, and I keep getting it wrong

**Reply by friedenhe** [reply]
> Your answer is very clear. Let me confirm again, do I only need to modify the `inlet type` in `0/U`? 

Yes

> And how do you choose the type of `outlet`? Because I've tried it before, and I keep getting it wrong

That depends on your problem...

**Reply by 101709** [reply]
Can I use it in DAPimpleFoam and timeAccurate mode?

**Reply by friedenhe** [reply]
@101709 Yes
